modules/roles/main.tf

```diff
@@ -0,0 +1,46 @@
+data "aws_caller_identity" "current" {}
+
+locals {
+  assume_role_policy = jsonencode(var.assume_role_policy != null ? var.assume_role_policy : {
+    Version = "2012-10-17",
+    Statement = [{
+      Action    = "sts:AssumeRole"
+      Effect    = "Allow"
+      Principal = { AWS = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:root" }
+    }]
+  })
+  policies = [
+    for name in var.policies :
+    lookup(var.policy_dict, name, name)
+  ]
+}
+
+resource "aws_iam_role" "main" {
+  force_detach_policies = true
+
+  assume_role_policy = local.assume_role_policy
+  description        = var.description
+  name               = var.name
+  tags               = var.tags
+}
+
+resource "aws_iam_role_policy_attachment" "main" {
+  for_each = { for k, v in local.policies : k => v }
```

Comment:
`local.policies` is a tuple, so it is not enough. If you don't like the way it is written then it can be changed to:

```tf
resource "aws_iam_role_policy_attachment" "main" {
  for_each = toset(local.policies)

  role       = aws_iam_role.main.name
  policy_arn = each.value
}
```

Reply:
O, right! maybe it's better to change local.policies then? Ie. to:
```hcl
policies = {
    for name in var.policies :
    name => lookup(var.policy_dict, name, name)
  }
```
then it's dict and you can use `for_each = local.policies` directly ;)

Reply:
good point, thx